server/index.js

```diff
@@ -61,6 +61,8 @@ app.get('/callback', (req, res, next) => {
         const queryParams = new URLSearchParams({ access_token, refresh_token, expires_in });
 
         res.redirect(`/#auth?${queryParams}`);
+      } else {
+        window.alert('A connection error occured, please try again');
```

Comment:
You won't be able to send up a window alert on the server. There's no window in Node.js. 

Instead, redirect the user to a different page that tells them that the authorization failed.